providers/nextcloud.go

```diff
@@ -22,3 +29,46 @@ func NewNextcloudProvider(p *ProviderData) *NextcloudProvider {
 	}
 	return &NextcloudProvider{ProviderData: p}
 }
+
+// EnrichSession is called after Redeem to allow providers to enrich session fields
+// such as User, Email, Groups with provider specific API calls.
+func (p *NextcloudProvider) EnrichSession(ctx context.Context, s *sessions.SessionState) error {
+	json, err := requests.New(p.ValidateURL.String()).
```

Comment:
Something I missed before, why is this `ValidateURL` rather than `ProfileURL`? I think if we are going to set this up as enriching from the profile we should make the `ProfileURL` required for this provider (and then not enrich if we don't have it set)